### PR TITLE
Add 4 process run for dry baroclinic wave

### PR DESCRIPTION
**Pull request by sriharshakandala**
## Purpose 
Add 4 process run for dry baroclinic wave

----
- [ ] I have read and checked the items on the review checklist.

### CODE REVIEW

File: config/gpu_configs/target_gpu_implicit_baroclinic_wave_4process.yml
Comment (simonbyrne) [inline config/gpu_configs/target_gpu_implicit_baroclinic_wave_4process.yml:1]: Can we just use the same config as the single process case?

File: .buildkite/gpu_pipeline/pipeline.yml [hunk 1]
```diff
@@ -50,3 +50,14 @@ steps:
         agents:
           slurm_gpus: 1
           slurm_time: 23:00:00
+
+      - label: "target_gpu_implicit_baroclinic_wave_4process"
+        command: >
+          julia --project -e 'using CUDA; CUDA.versioninfo()'
+          
+          nsys profile --trace=nvtx,cuda --output=target_gpu_implicit_baroclinic_wave_4process/report julia --color=yes --project=examples examples/hybrid/driver.jl
+          --config_file ${GPU_CONFIG_PATH}target_gpu_implicit_baroclinic_wave_4process.yml
```
Comment (simonbyrne) [inline .buildkite/gpu_pipeline/pipeline.yml:59]: I think you need an `srun` in there somewhere?

Comment (sriharshakandala) [reply .buildkite/gpu_pipeline/pipeline.yml:59]: The config file also contains the `job_id`, which is used as  a directory name for storing output!

File: .buildkite/gpu_pipeline/pipeline.yml [hunk 2]
```diff
@@ -50,3 +50,14 @@ steps:
         agents:
           slurm_gpus: 1
           slurm_time: 23:00:00
+
+      - label: "target_gpu_implicit_baroclinic_wave_4process"
+        command: >
+          julia --project -e 'using CUDA; CUDA.versioninfo()'
+          
+          srun julia --color=yes --project=examples examples/hybrid/driver.jl
```
Comment (simonbyrne) [inline .buildkite/gpu_pipeline/pipeline.yml:58]: Can we keep the `nsys profile`? You can use multi-line yaml to make it easier to read: https://github.com/CliMA/slurm-buildkite/wiki/Tips#splitting-commands-over-multiple-lines

File: .buildkite/gpu_pipeline/pipeline.yml [hunk 3]
```diff
@@ -44,9 +44,34 @@ steps:
...
+        agents:
+          slurm_gpus: 1
+          slurm_time: 23:00:00
+
+      - label: "target_gpu_implicit_baroclinic_wave_4process"
+        command: >
+          julia --project -e 'using CUDA; CUDA.versioninfo()'
+          
+          nsys profile --trace=nvtx,cuda --output=target_gpu_implicit_baroclinic_wave_4process/report
+          srun julia --color=yes --project=examples examples/hybrid/driver.jl
```
Comment (simonbyrne) [inline .buildkite/gpu_pipeline/pipeline.yml:72]:
```suggestion
          mpiexec julia --color=yes --project=examples examples/hybrid/driver.jl
```

File: config/gpu_configs/target_gpu_implicit_baroclinic_wave_4process.yml
Comment (sriharshakandala) [reply config/gpu_configs/target_gpu_implicit_baroclinic_wave_4process.yml:1]: The config file also contains the job_id, which is used as a directory name for storing output!

**Comment by simonbyrne** [comment]
I think squash this, then it's good to merge.

**Comment by sriharshakandala** [comment]
> I think squash this, then it's good to merge.

I can move the moist simulation to a different PR, since this is currently not working on the GPU!

**Comment by simonbyrne** [comment]
might as well leave it: we will need it at some point.

**Review by simonbyrne — APPROVED** [review]

**Pull request merged**